app/main.py

```diff
@@ -1 +1,53 @@
-# write your code here
+class SoftwareEngineer:
+    skills = []
+
+    def __init__(self, name: str = "") -> None:
+        self.name = name
+
+    def learn_skill(self, skill: str) -> None:
+        self.skills.append(skill)
+
+
+class FrontendDeveloper(SoftwareEngineer):
+    skills = SoftwareEngineer.skills + ["JavaScript", "HTML", "CSS"]
+
+    def __init__(self, name: str = "") -> None:
+        super().__init__(name)
+
+    def create_awesome_web_page(self) -> str:
+        print(f"{self.name} is creating a webpage...")
+        return "<h1>Hello world</h1>"
+
+
+class BackendDeveloper(SoftwareEngineer):
+    skills = SoftwareEngineer.skills + ["Python", "SQL", "Django"]
+
+    def __init__(self, name: str = "") -> None:
+        super().__init__(name)
+
+    def create_powerful_api(self) -> str:
+        print(f"{self.name} is creating an API...")
+        return "http://127.0.0.1:8000"
+
+
+class AndroidDeveloper(SoftwareEngineer):
+    skills = SoftwareEngineer.skills + ["Java", "Android studio"]
+
+    def __init__(self, name: str = "") -> None:
+        super().__init__(name)
+
+    def create_smooth_mobile_app(self) -> str:
+        print(f"{self.name} is creating a mobile app...")
+        return "Ads every three swipes"
+
+
+class FullStackDeveloper(FrontendDeveloper, BackendDeveloper):
+    skills = FrontendDeveloper.skills + BackendDeveloper.skills
```

Comment:
u don't need to hardcode skills like that
just extend skills in the constructor method